frontend/packages/dev-console/src/components/dropdown/ResourceDropdown.tsx

```diff
@@ -128,13 +138,16 @@ class ResourceDropdown extends React.Component<ResourceDropdownProps, State> {
 
   private onChange = (key: string) => {
     const name = this.state.items[key];
-    const { actionItem, onChange } = this.props;
-    const title = actionItem && key === actionItem.actionKey ? actionItem.actionTitle : name;
+    const { onChange } = this.props;
     onChange && this.props.onChange(key, name);
-    this.setState({ title });
   };
 
   render() {
+    const firstItemKey = _.get(_.keys(this.state.items), 0);
+    if (this.props.autoSelect && firstItemKey) {
+      this.onChange(firstItemKey);
```

Comment:
Never make a callback inside `render.

Reply:
Cannot even add it in `componentWillRecieveProps` as it exceeds the maximum state update depth. Trying to figure out some other way.

Reply:
You must be getting into a loop. You'll need to guard against setting on change to the same value as what's already set.

Reply:
Yes exactly, that is what happens if I add it in `componentWillRecieveProps`. To avoid that I have added a condition to check if the new title is different from what is already there in the state and if yes then call `onChange` and also set the new title in the state.